service/routes.py

```diff
@@ -70,3 +70,20 @@ def check_content_type(content_type):
         status.HTTP_415_UNSUPPORTED_MEDIA_TYPE,
         f"Content-Type must be {content_type}",
     )
+
+def list_products():
+    """Returns all of the Products"""
+    app.logger.info("Request for Product list")
+    products = []
+    category = request.args.get("category")
+    name = request.args.get("name")
+    if category:
+        products = Product.find_by_category(category)
+    elif name:
+        products = Product.find_by_name(name)
+    else:
+        products = Product.all()
```

Comment:
For the homework, the professor said we don't need line78 to line 84 right now, but Jacob write the function of find_by_name, you could try to adjust your code